File: app/src/main/java/com/cybersec/smartroute/storage/MockLocationSessionStore.java
package com.cybersec.smartroute.storage;

import android.content.Context;
import android.content.SharedPreferences;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Plain SharedPreferences slot for the active mock session. Kept separate
 * from the encrypted config store so the foreground service / boot receiver
 * can read it without touching keystore.
 */
public final class MockLocationSessionStore {

    private static final String PREFS = "smart_route_session";
    private static final String KEY_SESSION = "session_json";
    private static final String KEY_ACTIVE = "session_active";

    private final SharedPreferences prefs;

    public MockLocationSessionStore(Context context) {
        this.prefs = context.getApplicationContext()
                .getSharedPreferences(PREFS, Context.MODE_PRIVATE);
    }

    public void saveSession(JSONObject session) {
        prefs.edit()
                .putString(KEY_SESSION, session.toString())
                .putBoolean(KEY_ACTIVE, true)
                .apply();
    }

    public JSONObject loadSession() {
        if (!prefs.getBoolean(KEY_ACTIVE, false)) return null;
        String raw = prefs.getString(KEY_SESSION, null);
        if (raw == null) return null;
        try {
            return new JSONObject(raw);
        } catch (JSONException ex) {
            return null;
        }
    }

    public void clearSession() {
        prefs.edit()
                .remove(KEY_SESSION)
                .putBoolean(KEY_ACTIVE, false)
                .apply();
    }

    public boolean isSessionActive() {
        return prefs.getBoolean(KEY_ACTIVE, false);
    }
}

File: android/app/src/main/kotlin/com/cybersec/smartroute/MockLocationSessionStore.kt
package com.cybersec.smartroute

import android.content.Context
import org.json.JSONObject

/**
 * Persists active mock session so the foreground service survives process death.
 */
class MockLocationSessionStore(context: Context) {
    private val prefs =
        context.applicationContext.getSharedPreferences(PREFS, Context.MODE_PRIVATE)

    fun saveSession(json: JSONObject) {
        prefs.edit()
            .putString(KEY_SESSION, json.toString())
            .putBoolean(KEY_ACTIVE, true)
            .apply()
    }

    fun loadSession(): JSONObject? {
        if (!prefs.getBoolean(KEY_ACTIVE, false)) return null
        val raw = prefs.getString(KEY_SESSION, null) ?: return null
        return try {
            JSONObject(raw)
        } catch (_: Exception) {
            null
        }
    }

    fun clearSession() {
        prefs.edit()
            .remove(KEY_SESSION)
            .putBoolean(KEY_ACTIVE, false)
            .apply()
    }

    fun isSessionActive(): Boolean = prefs.getBoolean(KEY_ACTIVE, false)

    companion object {
        private const val PREFS = "smart_route_session"
        private const val KEY_SESSION = "session_json"
        private const val KEY_ACTIVE = "session_active"
    }
}
